File: crates/flowrt-codegen/tests/golden/iox2_service_rust/expected/app/stubs/rust/plan_service.rs
// FlowRT 管理参考模板（Rust）。可删除重建；复制到用户 app/ 后再修改。

#[derive(Default)]
pub struct PlanService;

impl flowrt_app::components::PlanService for PlanService {
    fn on_plan_request(
        &mut self,
        request: &flowrt_app::messages::PlanRequest,
    ) -> flowrt::ServiceResult<flowrt_app::messages::PlanResponse> {
        let _ = request;
        flowrt::ServiceResult::err(flowrt::ServiceError::HandlerError)
    }

    fn on_tick(&mut self) -> flowrt::Status {
        flowrt::Status::Ok
    }
}

pub fn build_app() -> flowrt_app::runtime_shell::App {
    flowrt_app::runtime_shell::App::new(Box::new(PlanService::default()))
}

File: crates/flowrt-codegen/tests/golden/bounded_service_iox2_cpp/expected/app/stubs/cpp/plan_service.cpp
// FlowRT 管理参考模板（C++）。可删除重建；复制到用户 app/ 后再修改。

#include "flowrt_app/runtime_shell.hpp"

#include <memory>

namespace {

class PlanService final : public flowrt_app::PlanServiceInterface {
public:
    flowrt::ServiceResult<flowrt_app::PlanResponse> on_plan_request(const flowrt_app::PlanRequest& request) override {
        (void)request;
        return flowrt::ServiceResult<flowrt_app::PlanResponse>::err(flowrt::ServiceError::HandlerError);
    }

    flowrt::Status on_tick() override {
        return flowrt::ok();
    }
};

}  // namespace

namespace flowrt_user {

flowrt_app::App build_app() {
    return flowrt_app::App(std::make_unique<PlanService>());
}

}  // namespace flowrt_user
